src/rulesets/LinearAlgebra/utils.jl

```diff
@@ -2,16 +2,18 @@
 # to any particular rule definition
 
 # F .* (X - X'), overwrites X if possible
-function _mulsubtrans!!(X::AbstractMatrix{T}, F::AbstractMatrix{T}) where T<:Real
+function _mulsubtrans!!(X::AbstractMatrix{<:Real}, F::AbstractMatrix{<:Real})
+    T = promote_type(eltype(X), eltype(F))
+    Y = (T == eltype(X)) ? X : similar(X, T)
```

Comment:
Not sure if there are many situations where this could happen, but just to be on the safe side.
```suggestion
    Y = (T <: eltype(X)) ? X : similar(X, T)
```